Source/WebCore/platform/graphics/controls/ControlPart.h

```diff
@@ -29,15 +29,15 @@
 #include "ControlFactory.h"
 #include "PlatformControl.h"
 #include "StyleAppearance.h"
-#include <wtf/RefCounted.h>
+#include <wtf/ThreadSafeRefCounted.h>
 
 namespace WebCore {
 
 class FloatRect;
 class GraphicsContext;
 class ControlFactory;
 
-class ControlPart : public RefCounted<ControlPart> {
+class ControlPart : public ThreadSafeRefCounted<ControlPart> {
```

Comment:
This makes sense. I checked the DisplayListItems.h and I found four ref counted objects stored by the items:
1. Gradient
2. Filter
3. ControlPart
4. SystemImage

The first two are sub-classes of `RenderingResource` which is `ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr`. So these are already thread safe. The last two are just `RefCounted`. So I would suggest making `SystemImage` `ThreadSafeRefCounted` also.

Reply:
Great, I will see if I can find a testcase using SystemImage. Or do you know any I could check?

Reply:
Oh, I cannot see any usage of this in non-Apple ports. But I'll still include the fix.